CodeEditModules/Modules/WorkspaceClient/src/Model/FileItem.swift

```diff
@@ -257,7 +257,12 @@ public extension WorkspaceClient {
             var fileUrl = self.url
             while FileItem.fileManger.fileExists(atPath: fileUrl.path) {
                 let previousName = fileUrl.lastPathComponent
-                fileUrl = fileUrl.deletingLastPathComponent().appendingPathComponent("\(previousName) copy")
+                let previousNameWithoutExtension = (previousName as NSString).deletingPathExtension
```

Comment:
`deletingPathExtension()` (note: an method) is also available at `URL` so i would avoid the additional casting to ~NSString

Reply:
I tried that like this -> `let previousNameWithoutExtension = fileUrl.deletingPathExtension().absoluteString`

But kept getting errors when trying to duplicate file saying -> `Thread 1: Fatal error: The file “xxx” doesn’t exist.`

Only way around it was doing the additional casting